projects/i18n/src/lib/services/i18n/i18n.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { Observable } from 'rxjs';
+
+import { AbstractI18nService } from '../../abstracts';
+
+@Injectable()
+export class I18nService implements AbstractI18nService {
+	constructor(private translateService: TranslateService) {}
+
+	public get currentLanguage(): string {
+		return this.translateService.currentLang;
+	}
+
+	public get availableLanguages(): string[] {
+		return this.translateService.langs;
+	}
+
+	public get defaultLanguage(): string {
+		return this.translateService.getDefaultLang();
+	}
+
+	public initI18n(language: string): Observable<unknown> {
+		this.translateService.use(language);
+
+		return this.translateService.reloadLang(language);
+	}
+
+	public setLanguage = (language: string): void => {
+		this.translateService.use(language);
+	};
+
+	public getTranslation(key: string, params?: any): string {
+		return this.translateService.instant(key, params);
+	}
+
+	// Denis: Set the default to string but allow for others.
+	public getTranslationObservable<GItemType = string>(
```

Comment:
Ik weet eigenlijk niet waarvoor de G in `GItemType` staat, dus mss beter geen afkorting van maken 😅